lib/rules/require-meta-type.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * @fileoverview require rules to implement a meta.type property
+ * @author 薛定谔的猫<weiran.zsd@outlook.com>
+ */
+
+'use strict';
+
+const utils = require('../utils');
+const validTypes = ['problem', 'suggestion', 'layout'];
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'require rules to implement a meta.type property',
+      category: 'Rules',
+      recommended: false, // TODO: enable it in a major release.
+    },
+    type: 'problem',
+    fixable: null,
+    schema: [],
+    messages: {
+      missing: '`meta.type` is required.',
```

Comment:
If someone is unfamiliar with `meta.type`, it might be useful to list the possible values in this message too. For example:

> \`meta.type\` is required (must be either \`problem\`, \`suggestion\`, or \`layout\`).